### PR TITLE
feat: handling of ucc modinput-functional tests

### PR DESCRIPTION
**Pull request by mkolasinski-splunk**
- logic for differentiating between old and new type of modinput functional tests
- generation and upload to aws swagger_client files neccesary for further test execution

Old type of tests run: 
https://github.com/splunk/splunk-add-on-for-microsoft-system-center-operations-manager/actions/runs/5539539317/jobs/10111342142
New type of tests run: 
https://github.com/splunk/splunk-add-on-for-google-workspace/actions/runs/5531119043/jobs/10111598916

Workaround applied for poetry bug as per:
python-poetry/poetry/issues/1917

### CODE REVIEW

File: .github/workflows/reusable-build-test-release.yml [hunk 1]
```diff
@@ -369,13 +369,23 @@ jobs:
       requirement_test: ${{ steps.testset.outputs.requirement_test }}
       scripted_inputs: ${{ steps.testset.outputs.scripted_inputs }}
       escu: ${{ steps.testset.outputs.escu }}
+      old_modinput_functional: ${{ steps.modinput-version.outputs.old_modinput_tests }}
```
Comment (hsekowski-splunk) [inline .github/workflows/reusable-build-test-release.yml:372]: Avoid names like 'old' or 'new'. When newer version is needed it ends with 'older', 'newer', 'older2' etc. names

File: .github/workflows/reusable-build-test-release.yml [hunk 2]
```diff
@@ -799,6 +816,38 @@ jobs:
...
+        if: steps.download-openapi.conclusion != 'skipped'
+        shell: bash
+        env:
+          PYTHON_KEYRING_BACKEND: keyring.backends.null.Keyring
+        run: |
+          sudo pip3 install poetry==1.2.2 
+          export POETRY_HTTP_BASIC_SPLUNK_ADD_ON_UCC_MODINPUT_TEST_USERNAME=${{ secrets.SA_GH_USER_NAME }}
+          export POETRY_HTTP_BASIC_SPLUNK_ADD_ON_UCC_MODINPUT_TEST_PASSWORD=${{ secrets.GH_TOKEN_ADMIN }}
+          poetry install --only modinput
+          poetry run ucc-test-modinput -o ${{ steps.download-openapi.outputs.download-path }}/openapi.json -t ${{ steps.download-openapi.outputs.download-path }}/tmp/
```
Comment (hsekowski-splunk) [inline .github/workflows/reusable-build-test-release.yml:840]: I'm happy to see the script usage :)

**Review by hsekowski-splunk — APPROVED** [review]
Just consider changing the 'old', 'new' name

File: .github/workflows/reusable-build-test-release.yml (hunk 1 above)
Comment (mkolasinski-splunk) [reply .github/workflows/reusable-build-test-release.yml]: Agree - resolved

File: .github/workflows/reusable-build-test-release.yml [hunk 3]
```diff
@@ -763,14 +776,18 @@ jobs:
...
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: recursive
+          token: ${{ secrets.GH_TOKEN_ADMIN }}
       - name: setup for test
         id: test-setup
         shell: bash
         run: |
+          sudo apt-get install -y crudini
```
Comment (artemrys) [inline .github/workflows/reusable-build-test-release.yml:790]: This was installed in the container image before?

File: .github/workflows/reusable-build-test-release.yml [hunk 4]
```diff
@@ -799,6 +816,38 @@ jobs:
...
+      - name: Setup python
+        if: steps.download-openapi.conclusion != 'skipped'
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.7
+      - name: modinput-test-prerequisites
+        if: steps.download-openapi.conclusion != 'skipped'
+        shell: bash
+        env:
+          PYTHON_KEYRING_BACKEND: keyring.backends.null.Keyring
```
Comment (artemrys) [inline .github/workflows/reusable-build-test-release.yml:834]: What is the purpose of this?

File: .github/workflows/reusable-build-test-release.yml (hunk 3 above)
Comment (mkolasinski-splunk) [reply .github/workflows/reusable-build-test-release.yml:790]: Yes - we need to use crudini so I am resolving this conv.

File: .github/workflows/reusable-build-test-release.yml (hunk 4 above)
Comment (mkolasinski-splunk) [reply .github/workflows/reusable-build-test-release.yml:834]: Yet another poetry bug: https://github.com/python-poetry/poetry/issues/1917

I will add this to PR description.

**Review by artemrys — APPROVED** [review]

**Pull request merged**